crates/pipeline-manager/src/runner.rs

```diff
@@ -324,11 +326,11 @@ impl RunnerApi {
         }
         let location = pipeline.deployment_location.unwrap(); // TODO: unwrap
 
-        // TODO: it might be better to have ?name={}, otherwise we have to
-        // restrict name format
         let url = format!("http://{location}/{endpoint}?{}", req.query_string());
 
-        let mut request = client.request(req.method().clone(), url);
+        let mut request = client
+            .request(req.method().clone(), url)
+            .timeout(timeout.unwrap_or(Self::PIPELINE_HTTP_REQUEST_TIMEOUT));
```

Comment:
What's the effect of this timeout? We may definitely want to stream for more than 30 seconds, e.g., when listening to the output stream of a pipeline.